homeassistant/components/shelly/config_flow.py

```diff
@@ -144,15 +153,19 @@ async def async_step_user(
                     )
                 except DeviceConnectionError:
                     errors["base"] = "cannot_connect"
+                except WrongShellyGen:
+                    errors["base"] = "wrong_shelly_gen"
                 except Exception:  # pylint: disable=broad-except
                     LOGGER.exception("Unexpected exception")
                     errors["base"] = "unknown"
                 else:
                     if device_info["model"]:
+                        host, port = await parse_host(user_input[CONF_HOST])
                         return self.async_create_entry(
                             title=device_info["title"],
                             data={
-                                **user_input,
+                                CONF_HOST: host,
+                                CONF_PORT: port,
```

Comment:
Should we increase the  config entry `minor_version` if we add a new field? https://developers.home-assistant.io/blog/2023/12/18/config-entry-minor-version/